xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -822,10 +822,7 @@ absl::Status IrEmitterUnnested::EmitCublasLtMatmulThunkF8(
   BufferAllocation::Slice aux;  // Not used.
   TF_RET_CHECK(!has_aux_output);
   std::optional<BufferAllocation::Slice> workspace_buffer;
-  if (instr->shape().IsTuple() &&
-      (instr->shape().tuple_shapes_size() - has_aux_output - 1)) {
-    TF_RET_CHECK((has_aux_output && instr->shape().tuple_shapes_size() == 3) ||
-                 (!has_aux_output && instr->shape().tuple_shapes_size() == 2));
+  if (instr->shape().tuple_shapes_size() - config.damax_output() == 2) {
```

Comment:
Is there a known case where this is false? If not, this is probably not needed.

Reply:
I don't know a case, but the workspace rewrite has a guard like this: 

      if (!((instr->shape().IsTuple() &&
             instr->shape().tuple_shapes_size() ==
                 has_aux_output + config.damax_output() + 1) ||
            instr->shape().IsArray())) {
       return absl::OkStatus();
}

If gemm_rewrite happens to create a new case where tuple includes other outputs, then it is an unsupported case.